src/rules/no-redundant-files.ts

```diff
@@ -0,0 +1,272 @@
+import type { Rule } from "eslint";
+import type { AST as JsonAST } from "jsonc-eslint-parser";
+import type { JSONStringLiteral } from "jsonc-eslint-parser/lib/parser/ast.js";
+
+import * as ESTree from "estree";
+
+import { createRule } from "../createRule.js";
+import { isJSONStringLiteral, isNotNullish } from "../utils/predicates.js";
+
+const defaultFiles = [
+	/* cspell:disable-next-line */
+	"LICENCE",
+	/* cspell:disable-next-line */
+	"LICENCE.md",
+	"LICENSE",
+	"LICENSE.md",
+	"package.json",
+	"README.md",
+] as const;
+
+const cachedRegex = new Map<string, RegExp>();
+const getCachedLocalFileRegex = (filename: string) => {
+	// Strip the leading `./`, if there is one, since we'll be incorporating
+	// it into the regex.
+	const baseFilename = filename.replace("./", "");
+	let regex = cachedRegex.get(baseFilename);
+	if (regex) {
+		return regex;
+	} else {
+		regex = new RegExp(`^(./)?${baseFilename}$`, "i");
+		cachedRegex.set(baseFilename, regex);
+		return regex;
+	}
+};
+
+/**
+ * A function to use as a fix, that will remove an element from an array of elements
+ * and manage the removal of any commas that would be necessary to maintain proper
+ * json.  If this is the last entry, and we don't have a comma (strict json), then
+ * we don't need to worry about removing this element's comma, but we should remove
+ * the comma from the element before, since it will the last element after the removal.
+ * If this is not the last line, and it's not the only line (which wouldn't have a comma)
+ * then we need to remove the comma from this line, so that it's not left behind after
+ * the node removal.
+ * @param elements All file entry elements
+ * @param currentIndex The index of the entry that we want to remove
+ * @param fixer The eslint fixer function we'll use to remove the entry
+ * @param context Current context
+ * @returns The list of fixes that the `context.report` fix needs to perform the removals for us
+ */
+const removeEntryAndAppropriateCommas = (
+	elements: JsonAST.JSONStringLiteral[],
+	currentIndex: number,
+	fixer: Rule.RuleFixer,
+	context: Rule.RuleContext,
+): Rule.Fix[] => {
+	const element = elements[currentIndex];
+	const fixes = [fixer.remove(element as unknown as ESTree.Node)];
```

Comment:
[Style] WDYT about switching this to a `function*`? It means we can get rid of a few lines of code & complexity with the array.

```ts
function* removeEntryAndAppropriateCommas( ... ) {
```

```suggestion
	yield fixer.remove(element as unknown as ESTree.Node);
```

I'm in favor, but open to being told this is annoying. 😄

Reply:
Updated